osrf/vrx
# [Garden] Delay the simulation exit

This pull request modifies the way we exit the simulation. Before, the simulation would terminate when:
  * `<per_plugin_exit_on_completion>` is set to `true` and the `remaining time` reaches `0`, or
  * `VRX_EXIT_ON_COMPLETION` environment variable is set to `true` and the `remaining time` reaches `0`, or
  * A plugin calls `Exit()`.
 
Now, the behavior is to terminate the simulation `<exit_time>` seconds after:
  * `<per_plugin_exit_on_completion>` is set to `true` and the `remaining time` reaches `0`, or
  * `VRX_EXIT_ON_COMPLETION` environment variable is set to `true` and the `remaining time` reaches `0`, or
  * A plugin calls `RequestExit()`.

~~I replaced `Exit()` with `RequestExit()` to make it a bit more explicit that you're requesting the termination but that won't happen right away.~~

Plugins should call `Finish()` to request the finish of the task. If needed, the simulation will also be terminated after the delay.

How to test it?

* Run the perception task:
```
ros2 launch vrx_gz competition.launch.py world:=perception_task
```

* Leave the simulation running and observe that simulation should terminate at second `42.0` (40 + 2).
